src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -254,7 +254,7 @@ private void WriteConstructorBody(CodeClass parentClass, CodeMethod currentMetho
                     conventions.AddParametersAssignment(writer,
                                                         pathParametersProp.Type,
                                                         string.Empty,
-                                                        (rawUrlParam.Type, Constants.RawUrlParameterName, rawUrlParam.Name.ToFirstCharacterLowerCase()));
+                                                        (rawUrlParam.Type, Constants.RawUrlParameterName, rawUrlParam.Name.ToFirstCharacterLowerCase(), rawUrlParam.Optional));
```

Comment:
Is there any reason why we would want the `rawUrlParam` to be nullable, as it is a constructor do we not want the user to explicitly pass a value that is not null and get an error if they do so?

Reply:
My assumption was that a required URL parameter wouldn't be optional as well and so both `rawUrlParam.Optional` and `rawUrlParam.Type.IsNullable` would be false. I hadn't thought about an optional required parameter with a default value. So, I've removed the optional check.